ocaml/xapi/repository_helpers.ml

```diff
@@ -731,15 +728,26 @@ module UpdateInfo = struct
                       | Xml.Element ("description", _, [Xml.PCData v]) ->
                           {acc with description= v}
                       | Xml.Element ("recommended_guidance", _, [Xml.PCData v])
-                        ->
-                          {acc with rec_guidance= Some (Guidance.of_string v)}
-                      | Xml.Element ("absolute_guidance", _, [Xml.PCData v]) ->
-                          {acc with abs_guidance= Some (Guidance.of_string v)}
+                        -> (
+                        try {acc with rec_guidance= Some (Guidance.of_string v)}
+                        with e ->
+                          (* The error should not block update. Ingore it. *)
+                          warn "%s" (ExnHelper.string_of_exn e) ;
+                          acc
+                      )
+                      | Xml.Element ("absolute_guidance", _, [Xml.PCData v])
+                        -> (
+                        try {acc with abs_guidance= Some (Guidance.of_string v)}
+                        with e ->
+                          (* The error should not block update. Ingore it. *)
+                          warn "%s" (ExnHelper.string_of_exn e) ;
```

Comment:
Might be useful to be able to report back any parsing errors in a separate part of the reply, probably something as a future enhancement. Then the cloud could keep track of these parsing failures so we eventually resolve them (or we could run a more strict version of this in xenrt to detect errors early).